tests/test_helpers_all_submissions.py

```diff
@@ -0,0 +1,78 @@
+"""Tests for UnauthenticatedReddit class."""
+
+from __future__ import print_function, unicode_literals
+
+import time
+
+from praw.helpers import all_submissions
+from .helper import PRAWTest, betamax
+
+
+def mock_time():
+    return 1448923624.0
+
+real_time = time.time
+
+
+class TestHelperAllSubmissions(PRAWTest):
+    def setUp(self):
+        PRAWTest.setUp(self)
+        time.time = mock_time
+
+    def tearDown(self):
+        PRAWTest.tearDown(self)
+        # make sure nose reports correct running time
+        time.time = real_time
+
+    @betamax()
+    def test_all_submissions(self):
+        try:
+            all_subs = list(all_submissions(self.r,
+                                            self.sr,
+                                            highest_timestamp=time.time(),
+                                            verbosity=3))
+
+            for i in range(len(all_subs) - 1):
+                self.assertGreaterEqual(all_subs[i].created_utc,
+                                        all_subs[i + 1].created_utc)
+
+            sr_obj = self.r.get_subreddit(self.sr)
+            all_subs_sr_object = list(all_submissions(self.r,
+                                                      sr_obj,
+                                                      verbosity=3))
+
+            def get_submissions_data(submissions_list):
+                return [(s.created_utc, s.title, s.url)
```

Comment:
Two lists of submissions should be directly comparable:

```
[<praw.objects.Submission at 0x110755ed0>,
 <praw.objects.Submission at 0x110755fd0>,
 <praw.objects.Submission at 0x110755150>,
 <praw.objects.Submission at 0x110755210>,
 <praw.objects.Submission at 0x1106c6b10>]

In [12]: b
Out[12]: 
[<praw.objects.Submission at 0x1106ca850>,
 <praw.objects.Submission at 0x1106bb0d0>,
 <praw.objects.Submission at 0x1106bb3d0>,
 <praw.objects.Submission at 0x1106bb290>,
 <praw.objects.Submission at 0x1106bb190>]

In [13]: a == b
Out[13]: True
```

Note that the objects are at different addresses, but their built-in comparison enables them to compare as equal if they're the same type and have the same id.